tests/integration/test_charm.py

```diff
@@ -27,21 +31,43 @@ async def test_build_and_deploy(ops_test: OpsTest, get_unit_ip_list) -> None:
     executable_jar_resources = {"spring-boot-app-image": "ghcr.io/canonical/spring-boot:3.0"}
     buildpack_resources = {"spring-boot-app-image": "ghcr.io/canonical/spring-boot:3.0-layered"}
     # Deploy the charm and wait for idle
-    app_name = "spring-boot-k8s"
-    buildpack_app_name = f"{app_name}-buildpack"
     await asyncio.gather(
         ops_test.model.deploy(
-            charm, resources=executable_jar_resources, application_name=app_name, series="jammy"
+            charm, resources=executable_jar_resources, application_name=APP_NAME, series="jammy"
         ),
         ops_test.model.deploy(
             charm,
             resources=buildpack_resources,
-            application_name=buildpack_app_name,
+            application_name=BUILDPACK_APP_NAME,
             series="jammy",
         ),
-        ops_test.model.wait_for_idle(apps=[app_name, buildpack_app_name], status="active"),
+        ops_test.model.wait_for_idle(apps=[APP_NAME, BUILDPACK_APP_NAME], status="active"),
     )
-    for name in [app_name, buildpack_app_name]:
+    for name in [APP_NAME, BUILDPACK_APP_NAME]:
         unit_ips = await get_unit_ip_list(name)
         for unit_ip in unit_ips:
             assert requests.get(f"http://{unit_ip}:8080/hello-world", timeout=5).status_code == 200
+
+
+@pytest.mark.abort_on_fail
+async def test_application_config_server_port(ops_test: OpsTest, get_unit_ip_list) -> None:
```

Comment:
Does the other case test the case where the post is not provided? Or maybe that is covered by a unit test?

Reply:
By default the port is not provided, so the previous test case has already covered that.